packages/apps/plugins/plugin-deck/src/DeckPlugin.tsx

```diff
@@ -182,7 +183,12 @@ export const DeckPlugin = ({
       },
       context: (props: PropsWithChildren) => {
         return (
-          <AttentionProvider attended={attention.attended}>
+          <AttentionProvider
+            attended={attention.attended}
+            onChangeAttend={(nextAttended) => {
+              attention.attended = nextAttended;
+            }}
```

Comment:
Consider updating this through an intent?

Reply:
I’m not sure it’s quite the same kind of event — this is connecting a UI component’s controllable state to a plugin’s `provides` state, which other plugins are free to access and subscribe to as a signal.

Reply:
I'll say that I've trended away from writing directly to signals even if they're exposed by provides. I've sometimes even made them readonly in the provides type to discourage that. It helps when changing any sort of logic around how the state is updated, there's just one place to do it.

For example, we used to update the `active` state like this, but that became problematic when I wanted to add `previous`.

Reply:
I suppose for this particular feature I’m of the (non-)opinion that this should change when we can demonstrate significant value can be delivered by changing it.

Currently, `attention` shares the same implementation as `location` in both `LayoutPlugin` and `DeckPlugin`, so when we decide these are better implemented another way, we can change both (or indeed all `provides`-related calls to `echo-schema`’s `create`) in one go.

Reply:
fwiw this is the only place `attended` should be updated. Maybe it should be read-only to consumers since the way that they “write” to this state is to `const attrs = useAttendable(id)` and `<div {...atts}>`. Do either of you anticipate any specific harm from leaving the current implementation as-is?

Reply:
I'm fine with leaving it as is, I'd maybe suggest adding a TODO as a reminder that if this in undergoing changes later that it probably warrants an intent.